### PR TITLE
Mention API and link to swagger

### PR DESCRIPTION
**Pull request by matkuliak**
## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+===
+API
+===
+
+Similarly to Croud, we offer API to allow programmatic access to our Cloud
+products. API is operated using a key and a secret. These can be created in
```
Comment (SStorm) [inline docs/api.rst:6]: I would perhaps phrase it as "The API can be accessed by generating a key and secret in your account page"

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- Regions
+- Projects
+- Clusters
+- Products
+- Users
+- Roles
+- Subscriptions
+- Audit logs
+- Metadata
+- Free trial credits
```
Comment (SStorm) [inline docs/api.rst:31]: I would perhaps not list Free trial credits and metadata here, even though they are available - they do not make much sense to the user :)

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Similarly to Croud, we offer API to allow programmatic access to our Cloud
+products. API is operated using a key and a secret. These can be created in
+your account page:
+
+.. image:: _assets/img/create-api-key.png
+   :alt: Cloud Console New Api Key
+
+Click the *Generate new key* button to create your key. A popup with your
+key and secret will appear. Make sure to store your secret safely, as you
+cannot access it again.
```
Comment (SStorm) [inline docs/api.rst:14]: I would also add that the key and secret can be used as HTTP Basic Auth credentials when calling the API. Maybe even worth giving an example:
```shell
$ curl -s -u $your_key:$your_secret https://console.cratedb.cloud/api/v2/users/me/
{"email":"some@example.com","hmac":"...","is_superuser":false,"name":"Some User","organization_id":"123","status":"active","uid":"uid","username":"some@example.com"}
```

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
+===
+API
+===
+
+Similarly to Croud, we offer API to allow programmatic access to our Cloud
```
Comment (kojinkai) [inline docs/api.rst:5]:
```suggestion
Similar to Croud, we offer an API to allow programmatic access to our Cloud
```

**Review by SStorm — APPROVED** [review]

**Review by SStorm — APPROVED** [review]

**Pull request merged**